python/toolbox/system_cpu_topology.py

```diff
@@ -0,0 +1,273 @@
+import copy
+from pathlib import Path
+
+def _debug_log(log_msg):
+    return(print("DEBUG: %s" % (log_msg)))
+
+class system_cpu:
+    def __init__(self, cpu_dir, debug = False):
+        self.debug = debug
+
+        if self.debug:
```

Comment:
if you use logger, you don't need these checks (if debug:)...
    logformat = '%(asctime)s %(levelname)s %(name)s:  %(message)s'

You only check once:
    if debug:
        logging.basicConfig(level=logging.DEBUG, format=logformat)
    else:
        logging.basicConfig(level=logging.INFO, format=logformat)
    log = logging.getLogger(__name__)

Then you can freely use log.info("hey this is info msg") or log.debug("hey this is debug msg") without any if else block.

log.info("info msg is shown by default as we set in basicConfig")
log.debug("this will print only if if debug is enabled")

Reply:
I have switched over to this technique with a few tweaks in certain places.